middleware/logging.go

```diff
@@ -53,18 +58,31 @@ func (l Log) Wrap(next http.Handler) http.Handler {
 
 			return
 		}
-		if 100 <= statusCode && statusCode < 500 || statusCode == http.StatusBadGateway || statusCode == http.StatusServiceUnavailable {
+		if 100 <= statusCode && statusCode < 500 {
 			l.logWithRequest(r).Debugf("%s %s (%d) %s", r.Method, uri, statusCode, time.Since(begin))
 			if l.LogRequestHeaders && headers != nil {
 				l.logWithRequest(r).Debugf("ws: %v; %s", IsWSHandshakeRequest(r), string(headers))
 			}
+		} else if statusCode == http.StatusBadGateway || statusCode == http.StatusServiceUnavailable {
+			l.logHighVolumeError(r, "%s %s (%d) %s", r.Method, uri, statusCode, time.Since(begin))
+			if l.LogRequestHeaders && headers != nil {
+				l.logHighVolumeError(r, "ws: %v; %s", IsWSHandshakeRequest(r), string(headers))
```

Comment:
do you think we want this on a different line to the method?
(I see it did this in the code before, but only at debug level)

Reply:
i think it's ok.  this way the user can configure `LogRequestHeaders` and `HighVolumeErrorLog` independently and any combination works.

Reply:
What I'm thinking is: suppose you want to look at logs for a specific request; you can grep for the URI.  But the headers are on the next line, which is awkward.

Reply:
oh, that's a good point.  i hadn't considered that.  i'll push a small change to unify them